src/java.base/share/classes/jdk/internal/foreign/abi/AbstractLinker.java

```diff
@@ -101,22 +103,26 @@ public SystemLookup defaultLookup() {
         return SystemLookup.getInstance();
     }
 
+    /** {@return byte order used by this linker} */
+    protected abstract ByteOrder linkerByteOrder();
+
     // Current limitation of the implementation:
     // We don't support packed structs on some platforms,
     // so reject them here explicitly
-    private static void checkHasNaturalAlignment(FunctionDescriptor descriptor) {
-        descriptor.returnLayout().ifPresent(AbstractLinker::checkHasNaturalAlignmentRecursive);
-        descriptor.argumentLayouts().forEach(AbstractLinker::checkHasNaturalAlignmentRecursive);
+    private void checkLayouts(FunctionDescriptor descriptor) {
+        descriptor.returnLayout().ifPresent(this::checkLayoutsRecursive);
+        descriptor.argumentLayouts().forEach(this::checkLayoutsRecursive);
```

Comment:
Storing `this::checkLayoutsRecursive` in a local variable allows the `Consumer` to be reused for the return and argument layouts, thus only one lambda class and instance needs to be allocated instead of two.
```suggestion
        final Consumer<MemoryLayout> checker = this::checkLayoutsRecursive;
        descriptor.returnLayout().ifPresent(checker);
        descriptor.argumentLayouts().forEach(checker);
```